packages/commonwealth/server/passport/magic.ts

```diff
@@ -337,10 +322,38 @@ async function loginExistingMagicUser({
       );
     }
 
-    // TODO: we should also remove ghost addresses her, per code in /updateAddress...
-    // or we can leave it alone, although it wont migrate address ownership, most parameters
-    // are by profile anyway.
+    // remove ghost addresses once replacements have been generated
+    const ghostAddresses =
+      existingUserInstance?.Addresses?.filter(
+        ({ ghost_address }: AddressAttributes) => !!ghost_address,
+      ) || [];
+    for (const ghost of ghostAddresses as AddressAttributes[]) {
+      const replacementAddress = addressInstances.find(
+        ({ community_id, ghost_address }: AddressAttributes) =>
+          !ghost_address && community_id === ghost.community_id,
+      );
 
+      // should always exist, but check for it to avoid null check error
+      if (replacementAddress) {
+        // update data objects and delete ghost address
+        await models.Comment.update(
```

Comment:
It should not have Collaborations or SsoTokens, because the address will only have data that was migrated from discourse, which doesn't include any of these. Memberships should be removed and re-created if needed (@timolegros please confirm they get removed on cascading delete of the Address), because the address itself is (a) randomized and (b) changing.

Reply:
Memberships are not cascade deleted when an address is deleted:
```
SELECT
    conname AS constraint_name,
    CASE confdeltype
        WHEN 'a' THEN 'NO ACTION'
        WHEN 'r' THEN 'RESTRICT'
        WHEN 'c' THEN 'CASCADE'
        WHEN 'n' THEN 'SET NULL'
        WHEN 'd' THEN 'SET DEFAULT'
        END AS on_delete_action
FROM
    pg_constraint
WHERE
    conname = 'Memberships_address_id_fkey';
```

This query shows that the fkey on `Memberships.address_id` has `ON DELETE NO ACTION`

Can we guarantee that a collaboration won't exist? For example, what if we migrate a community from Discourse and one user signs in and sets the ghost address of another user as a collaborator before that user can sign in (and remove their own ghost address) - this transaction would fail due to foreign key constraint when you try to delete the ghost address (no cascade delete here either).

Reply:
Good catch.